### PR TITLE
Fix more upstream objc provider changes

### PR DESCRIPTION
**Pull request by keith**
This also switches us back to real last_green CI. The other linked issue is fixed at head

This reverts commit 9c0c81a174ef2ecaf8833ac0bf770b5328e13edd.

Fixes https://github.com/bazelbuild/rules_apple/issues/2370

### CODE REVIEW

**Review by luispadron — APPROVED** [review]

**Review by BalestraPatrick — APPROVED** [review]

**Review by mattrobmattrob — APPROVED** [review]

File: apple/internal/linking_support.bzl [hunk 1]
```diff
@@ -114,7 +110,12 @@ def _sectcreate_objc_provider(label, segname, sectname, file):
                 ]),
             ),
         ),
-    ]
+    ] + ([
+        apple_common.new_objc_provider(
+            linkopt = depset(linkopts, order = "topological"),
+            link_inputs = depset([file]),
+        ),
+    ] if _OBJC_PROVIDER_LINKING else [])
```
Comment (brentleyjones) [inline apple/internal/linking_support.bzl:118]: If this is needed, can it be its own PR?
